spinalcordtoolbox/centerline/curve_fitting.py

```diff
@@ -14,16 +14,15 @@
 
 def polyfit_1d(x, y, xref, deg=5):
     """
-    1d Polynomial fitting using np.polyfit
+    1d Polynomial fitting using numpy's Polynomial.fit
     :param x:
     :param y:
     :param deg:
     :param xref: np.array: vector of abscissa on which to project the fitted curve. Example: np.linspace(0, 50, 51)
     :return: p(xref): Fitted polynomial for each xref point
     :return: p.deriv(xref): Derivatives for each xref point
     """
-    from numpy import poly1d, polyfit
-    p = poly1d(polyfit(x, y, deg=deg))
+    p = np.polynomial.Polynomial.fit(x, y, deg)
```

Comment:
There is actually specified in [np.polyfit documentation](https://numpy.org/doc/1.18/reference/generated/numpy.polyfit.html):
`The Polynomial.fit class method is recommended for new code as it is more stable numerically.`

Makes sense :-)